src/twitch.js

```diff
@@ -32,30 +34,53 @@ Twitch.prototype.makeRequest = function(http) {
 }
 
 /**
-* @description : gets user data from the api
-* @param {String} username : the username we want information from
-* @returns {Promise.<string, Error>} : resolves JSON data or rejects an error
-*/
-Twitch.prototype.getUser = function(username) {
+ * @method getUserId
+ * @description : gets user id from username
+ * @param {String} username
+ * @returns {Promise.<string, Error>}
+ */
+Twitch.prototype.getUserId = function(username) {
     return new Promise((resolve, reject) => {
-        // set our URL for working with the api
-        let url = `https://api.twitch.tv/kraken/streams/${username}`;
-        // make our request
+        // set URL for working with the api
+        let url = `https://api.twitch.tv/kraken/users?login=${username}`;
+        // make request
         this.makeRequest(url)
             .then(data => {
-                // resolve our data and parse as a JSON
-                resolve(JSON.parse(data));
-            })
-            .catch(reject);
+                let json = JSON.parse(data);
+                resolve(json.users[0]._id);
```

Comment:
Not actually sure. I want to enforce strict type checks as well, but I'm not sure how that would hinder performance